sites/docs/pages/components/custom-components/component-queries.md

```diff
@@ -0,0 +1,406 @@
+---
+sidebar_position: 100
+title: Component Queries
+---
+
+Custom components in Evidence can include their own queries, allowing you to fetch data directly within the component rather than passing it as props. This gives you more flexibility in how you structure your components and can make them more self-contained.
+
+## Basic Query Implementation
+
+To add a query to your component, you'll need two key imports from Evidence's utilities:
+
+```javascript
+import { buildQuery } from '@evidence-dev/component-utilities/buildQuery';
+import { QueryLoad, Skeleton } from '@evidence-dev/core-components';
+```
+
+Here's a simple example of how to use these imports:
+
+```html title="components/TableList.svelte"
+&lt;script>
+    // Build a static query
+    const query = buildQuery(
+        'SELECT * FROM information_schema.tables',
+        'table-list-query' // Optional unique identifier
+    );
+</script>
+
+<QueryLoad data={query} let:loaded={tables}>
+    <ul>
+        {#each tables as table}
+            <li>{table.table_name}</li>
+        {/each}
+    </ul>
+</QueryLoad>
+```
+
+Let's break down what's happening:
+
+1. First, we use `buildQuery` to create our query. It takes two parameters:
+   - The SQL query text
+   - An optional unique identifier for the query
+
+2. We then use the `QueryLoad` component to handle the query execution and loading states. The `let:loaded` syntax gives us access to our query results once they're ready.
+
+<Alert status="info">
+
+**About Query IDs**
+
+While query IDs are optional, they can be helpful for debugging and performance optimization. If you're using multiple queries in your component, it's recommended to give each a unique identifier.
+
+</Alert>
+
+## Working with Loading States
+
+The `QueryLoad` component automatically handles loading states for your queries. You have three main approaches for managing how loading states appear:
```

Comment:
not super clear why you would pick between these at the moment.

The recommended option is also the most complex because you have to specify the dimensions of the slot. 

Why is the "Default" state not the recommended option